app/components-react/windows/go-live/useGoLiveSettings.ts

```diff
@@ -1,294 +1,213 @@
+      tweetText: view.getTweetText(view.commonFields.title), // generate a default tweet text
+      needPrepopulate: false,
+    };
+    // if stream has not been started than we allow to change settings only for a primary platform
+    // so delete other platforms from the settings object
+    if (this.isUpdateMode && !view.isMidStreamMode) {
+      Object.keys(settings.platforms).forEach((platform: TPlatform) => {
+        if (!this.checkPrimaryPlatform(platform)) delete settings.platforms[platform];
+      });
+    }
+    this.updateSettings(settings);
+  }
 
-  // create a TState shortcut
-  type TState = IGoLiveSettingsState;
+  getView(state: IGoLiveSettingsState) {
+    return new StreamInfoView(state);
+  }
+  get settings() {
+    return this.state;
+  }
 
-  // creates a StreamInfoView object based on provided state
-  function getView(state?: TState) {
-    const stateGetter = state ? () => state : getState;
-    return new StreamInfoView(StreamingService.state, stateGetter);
+  getSettings() {
+    return this.state;
   }
 
-  // create a StreamInfoView object based on the current state
-  const view = getView();
+  get isLoading() {
+    const state = this.state;
+    return state.needPrepopulate || this.getView(this.state).isLoading;
+  }
 
-  // DEFINE GETTERS
-  // getter must not have side effects
-  const getters = {
-    get isLoading() {
-      const state = getState();
-      return state.needPrepopulate || getView(state).isLoading;
-    },
+  get customDestinations() {
+    return this.state.customDestinations;
+  }
 
-    // antd form instance
-    get form() {
-      return form;
-    },
+  get optimizedProfile() {
+    return this.state.optimizedProfile;
+  }
 
-    // return last actual settings
-    getSettings() {
-      return getState();
-    },
+  get tweetText() {
+    return this.state.tweetText;
+  }
 
-    // select eligible layout and renders settings
-    renderPlatformSettings(
-      commonFields: JSX.Element,
-      requiredFields: JSX.Element,
-      optionalFields?: JSX.Element,
-      essentialOptionalFields?: JSX.Element,
-    ) {
-      let settingsMode: 'singlePlatform' | 'multiplatformAdvanced' | 'multiplatformSimple';
-      if (view.isMultiplatformMode) {
-        settingsMode = view.isAdvancedMode ? 'multiplatformAdvanced' : 'multiplatformSimple';
-      } else {
-        settingsMode = 'singlePlatform';
-      }
+  get isUpdateMode() {
+    return this.state.isUpdateMode;
+  }
 
-      switch (settingsMode) {
-        case 'singlePlatform':
-          return [essentialOptionalFields, commonFields, requiredFields, optionalFields];
-        case 'multiplatformSimple':
-          return requiredFields;
-        case 'multiplatformAdvanced':
-          return [essentialOptionalFields, requiredFields, optionalFields, commonFields];
-      }
-    },
-  };
+  /**
+   * Update top level settings
+   */
+  @mutation()
+  updateSettings(patch: Partial<IGoLiveSettingsState>) {
```

Comment:
are we ditching screaming snake  case for mutations with this switch? if so, what's the thinking?